tests/test_report.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+from src.report import (
+    get_report,
+    read_portfolio,
+    read_prices,
+    calc_gain_loss,
+    make_report,
+)
+from src import DATA_DIRECTORY
+
+
+def test_read_portfolio():
+    """
+    Tests if read_portfolio constructs each entry in the list correctly.
+    """
+    portfolio = read_portfolio(DATA_DIRECTORY / "portfolio.csv")
+    expected = [
+        {"name": "AA", "shares": 100, "price": 32.2},
+        {"name": "IBM", "shares": 50, "price": 91.1},
+        {"name": "CAT", "shares": 150, "price": 83.44},
+        {"name": "MSFT", "shares": 200, "price": 51.23},
+        {"name": "GE", "shares": 95, "price": 40.37},
+        {"name": "MSFT", "shares": 50, "price": 65.1},
+        {"name": "IBM", "shares": 100, "price": 70.44},
+    ]
+
+    assert portfolio == expected
+
+
+def test_read_portfolio_missing_values():
+    """
+    Tests if read_portfolio raises a ValueError if there is a blank line in the file.
+    """
+    with pytest.raises(ValueError):
+        read_portfolio("data/missing.csv")
```

Comment:
To be consistent you should use the `DATA_DIRECTORY` each time:
```suggestion
        read_portfolio(DATA_DIRECTORY / "missing.csv")
```